model/post_list.go

```diff
@@ -80,6 +81,12 @@ func (o *PostList) ToJSON() (string, error) {
 	return string(b), err
 }
 
+func (o *PostList) EncodeJSON(w io.Writer) error {
+	copy := *o
```

Comment:
This is a shallow copy. So it doesn't really do what you want. Let's just remove this one and the one below like above.